### PR TITLE
Feature/issue #15 login and role controllers

### PR DESCRIPTION
**Pull request by PedenkoIllia**
Tested JWT and authorization. All work correct. 
I took JWT from: https://jasonwatmore.com/post/2019/10/16/aspnet-core-3-role-based-authorization-tutorial-with-example-api

### CODE REVIEW

File: TimeOffTracker.WebApi/Controllers/RoleController.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            User user = await _userManager.FindByIdAsync(userId);
+            if (user != null)
+            {
+                try
+                {
+                    var userRole = await _userManager.GetRolesAsync(user);
+
+                    await _userManager.AddToRoleAsync(user, role);
+
+                    await _userManager.RemoveFromRolesAsync(user, userRole);
```
Comment (ASbeletsky) [inline TimeOffTracker.WebApi/Controllers/RoleController.cs]: what for?

File: TimeOffTracker.WebApi/Controllers/RoleController.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Cannot change user role: {UserId}", userId);
```
Comment (ASbeletsky) [inline TimeOffTracker.WebApi/Controllers/RoleController.cs]: don't use try catch manually - you have it in exception handler. Better to throw Exception(like InvalidOperation or create a custom exception) with this message. Error handler will handle log it, then retrun BadRequest

File: TimeOffTracker.WebApi/Services/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            var key = Encoding.ASCII.GetBytes(_appSettings.Secret);
+            string userRole = _userManager.GetRolesAsync(user).Result.FirstOrDefault();
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new Claim[]
+                {
+                    new Claim(ClaimTypes.Name, user.Id.ToString()),
+                    new Claim(ClaimTypes.Role, userRole)
+                }),
+                Expires = DateTime.UtcNow.AddDays(1),
```
Comment (ASbeletsky) [inline TimeOffTracker.WebApi/Services/UserService.cs]: move token lifetime to appsettings file

**Review by ASbeletsky — CHANGES_REQUESTED** [review]

File: TimeOffTracker.WebApi/Controllers/RoleController.cs (hunk 1 above)
Comment (PedenkoIllia) [reply TimeOffTracker.WebApi/Controllers/RoleController.cs]: I remove the old role after adding a new one

**Review by ASbeletsky — APPROVED** [review]

**Pull request merged**